community/bolt/src/test/java/org/neo4j/bolt/v1/messaging/Neo4jPackTest.java

```diff
@@ -180,4 +202,37 @@ public void shouldNotBeAbleToUnpackPaths() throws IOException
             unpacked( packed( path ) );
         }
     }
+
+    @Test
+    public void shouldTreatSingleCharAsSingleCharacterString() throws IOException
+    {
+        // Given
+        PackedOutputArray output = new PackedOutputArray();
+        Neo4jPack.Packer packer = new Neo4jPack.Packer( output );
+        packer.pack( 'C' );
+        Object unpacked = unpacked( output.bytes() );
+
+        // Then
+        assertThat( unpacked, instanceOf( String.class ) );
+        assertThat( unpacked, equalTo( "C" ) );
+    }
+
+    @Test
+    public void shouldTreatCharArrayAsListOfStrings() throws IOException
+    {
+        // Given
+        PackedOutputArray output = new PackedOutputArray();
+        Neo4jPack.Packer packer = new Neo4jPack.Packer( output );
+        packer.pack( new char[]{'W', 'H', 'Y'} );
+        Object unpacked = unpacked( output.bytes() );
+
+        // Then
+        assertThat( unpacked, instanceOf( List.class ) );
+        assertThat( unpacked, equalTo( asList( "W", "H", "Y" ) ) );
```

Comment:
So the final decision is to save char[] as string[]. (Thinking face but without any real opinion)